From 3b26ec436c150d4ed157fd5c017c4302d24197e6 Mon Sep 17 00:00:00 2001
From: Radu Pirea <radu-nicolae.pirea@nxp.com>
Date: Mon, 14 Dec 2020 10:03:32 +0200
Subject: [PATCH 32/76] s32: enable gpio-s32 driver and cmd_gpio

Enable by default the s32 gpio driver and cmd gpio on all the s32-gen1
boards.

Issue: ALB-5669

Signed-off-by: Radu Pirea <radu-nicolae.pirea@nxp.com>
---
 arch/arm/include/asm/gpio.h | 7 ++++++-
 board/freescale/Kconfig     | 3 +++
 drivers/net/dwc_eth_qos.h   | 4 +++-
 3 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/arch/arm/include/asm/gpio.h b/arch/arm/include/asm/gpio.h
index 333e407b66..7c00c2268f 100644
--- a/arch/arm/include/asm/gpio.h
+++ b/arch/arm/include/asm/gpio.h
@@ -1,3 +1,8 @@
+/* SPDX-License-Identifier: GPL-2.0+ */
+/*
+ * Copyright 2017-2021 NXP
+ */
+
 #if !defined(CONFIG_ARCH_UNIPHIER) && !defined(CONFIG_ARCH_STI) && \
 	!defined(CONFIG_ARCH_K3) && !defined(CONFIG_ARCH_BCM68360) && \
 	!defined(CONFIG_ARCH_BCM6858) && !defined(CONFIG_ARCH_BCM63158) && \
@@ -6,7 +11,7 @@
 	!defined(CONFIG_ARCH_LS1088A) && !defined(CONFIG_ARCH_ASPEED) && \
 	!defined(CONFIG_ARCH_LS1012A) && !defined(CONFIG_ARCH_LS1043A) && \
 	!defined(CONFIG_ARCH_LS1046A) && !defined(CONFIG_ARCH_U8500) && \
-	!defined(CONFIG_CORTINA_PLATFORM)
+	!defined(CONFIG_CORTINA_PLATFORM) && !defined(CONFIG_ARCH_S32)
 #include <asm/arch/gpio.h>
 #endif
 #include <asm-generic/gpio.h>
diff --git a/board/freescale/Kconfig b/board/freescale/Kconfig
index 36adacfe5d..8d1f252a1d 100644
--- a/board/freescale/Kconfig
+++ b/board/freescale/Kconfig
@@ -4,9 +4,11 @@ config S32V2
 
 config S32_GEN1
 	bool "S32 GEN1 Common Chassis"
+	imply CMD_GPIO
 	select ARM_SMCCC
 	select CLK
 	select CLK_SCMI
+	select DM_GPIO
 	select OF_CONTROL
 	select OF_SEPARATE
 	select PINCTRL
@@ -14,6 +16,7 @@ config S32_GEN1
 	select PINCTRL_S32
 	select SCMI_FIRMWARE
 	select ARCH_MISC_INIT
+	select S32_GPIO
 	help
 	  Platform Selection
 
diff --git a/drivers/net/dwc_eth_qos.h b/drivers/net/dwc_eth_qos.h
index 8ed541897b..da16b69a37 100644
--- a/drivers/net/dwc_eth_qos.h
+++ b/drivers/net/dwc_eth_qos.h
@@ -1,10 +1,12 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
  * Copyright (c) 2016, NVIDIA CORPORATION.
- * Copyright 2019-2020 NXP
+ * Copyright 2019-2021 NXP
  *
  */
 
+#include <asm-generic/gpio.h>
+
 #ifndef DWC_ETH_QOS_H
 #define DWC_ETH_QOS_H
 
-- 
2.17.1

